hyperspy/io.py

```diff
@@ -363,48 +399,68 @@ def load(filenames=None,
 
 
 def load_single_file(filename, **kwds):
-    """
-    Load any supported file into an HyperSpy structure
+    """Load any supported file into an HyperSpy structure.
+
     Supported formats: netCDF, msa, Gatan dm3, Ripple (rpl+raw),
     Bruker bcf, FEI ser and emi, EDAX spc and spd, hspy (HDF5), and SEMPER unf.
 
     Parameters
     ----------
     filename : string
-        File name (including the extension)
+        File name including the extension.
+    **kwds
+        Keyword arguments passed to specific file reader.
+
+    Returns
+    -------
+    object
+        Data loaded from the file.
 
     """
     if not os.path.isfile(filename):
         raise FileNotFoundError(f"File: {filename} not found!")
 
-    extension = os.path.splitext(filename)[1][1:]
-    i = 0
-
-    while extension.lower() not in io_plugins[i].file_extensions and \
-            i < len(io_plugins) - 1:
-        i += 1
-
-    if i == len(io_plugins):
-        # Try to load it with the python imaging library
-        try:
-            from hyperspy.io_plugins import image
-            reader = image
-            return load_with_reader(filename, reader, **kwds)
-        except BaseException as e:
-            raise IOError(
-                "If the file format is supported, "
-                f"please report this error:\n{e}"
-            )
+    # File extension without "." separator
+    file_ext = os.path.splitext(filename)[1][1:]
+    reader = kwds.pop("reader", None)
```

Comment:
This keyword needs to be added to the docstring of the `load` function. A quick example in this there would be good too.

Reply:
Done!